A66-otel-stats.md

```diff
@@ -0,0 +1,542 @@
+
+```
+public static class OpenTelemetryModuleBuilder {
+    /**
+     * OpenTelemetry instance is used to configure metrics settings.
+     *
+     * Sample
+     *    SdkMeterProvider sdkMeterProvider = SdkMeterProvider.builder()
+     *         .registerMetricReader(
+     *             PeriodicMetricReader.builder(
+     *                OtlpGrpcMetricExporter.builder().build()).build())
+     *         .build();
+     *
+     *     OpenTelemetry openTelemetry = OpenTelemetrySdk.builder()
+     *         .setMeterProvider(sdkMeterProvider)
+     *         .build();
+     *
+     * If MeterProvider is not configured, no-op meterProvider will be used by default.
+     * It provides meters which do not record or emit.
+     */
+    public OpenTelemetryModuleBuilder openTelemetry(OpenTelemetry openTelemetry);
+
+    /* Enable metrics for listed metrics. */
+    public OpenTelmetryBuilder enableMetrics(Set<String> metricNames);
+
+    /* If targetFilter returns true for a target, target is recorded as is.
+     * Otherwise it will be recorded as "other". */
+    public OpenTelemetryBuilder targetFilter(Predicate<String> targetFilter);
+
+    public OpenTelemetryModule build();
+}
+```
+
+#### Go
+
+```
+import (
+  "go.opentelemetry.io/otel/attribute"
+  "go.opentelemetry.io/otel/metric"
+)
+
+
+package opentelemetry
+
+// MetricsOptions are the metrics options for OpenTelemetry instrumentation.
+type MetricsOptions struct {
+  // MeterProvider is the MeterProvider instance that will be used for access
+  // to Named Meter instances to instrument an application. To enable metrics
+  // collection, set a meter provider. If unset, no metrics will be recorded.
+  // Any implementation knobs (i.e. views, bounds) set in the passed in object
+  // take precedence over the API calls from the interface in this component
+  // (i.e. it will create default views for unset views).
+  MeterProvider metric.MeterProvider
+
+  // Metrics are the metrics to instrument. Will turn on the corresponding
+  // metric supported by the client and server instrumentation components if
+  // applicable.
+  Metrics []string
+
+  // Attributes are constant attributes applied to every recorded metric.
+  Attributes []attribute.KeyValue
+}
+
+// DialOption returns a dial option which enables OpenTelemetry instrumentation
+// code for a grpc.ClientConn.
+//
+// Client applications interested in instrumenting their grpc.ClientConn should
+// pass the dial option returned from this function as a dial option to
+// grpc.Dial().
+//
+// For the metrics supported by this instrumentation code, a user needs to
+// specify the client metrics to record in metrics options. A user also needs to
+// provide an implementation of a MeterProvider. If the passed in Meter Provider
+// does not have the view configured for an individual metric turned on, the API
+// call in this component will create a default view for that metric.
+func DialOption(mo MetricsOptions) grpc.DialOption {}
```

Comment:
`DialOptionOptions`?  😆 

Seriously though I think we'll want this to take a struct that contains `MetricsOptions` instead so that we can include tracing options in the future.  Or a name besides `MetricsOptions` that includes tracing options as well.  Probably just `Options`?

Reply:
Yeah, I will scale this up to Options { metrics options, tracing options } (OpenCensus just had tracing options). Or should I change this and scale upward to not break users eventually (i.e. have an options struct with metrics options as a knob but no tracing options until tracing)?